lib/models/ReplicationConfiguration.js

```diff
@@ -281,30 +280,28 @@ class ReplicationConfiguration {
      * @return {undefined}
      */
     _parseStorageClass(destination) {
-        const storageClass = destination.StorageClass &&
-            destination.StorageClass[0];
-        if (!this._config) {
-            return validStorageClasses.includes(storageClass);
+        // StorageClass is optional.
+        if (destination.StorageClass === undefined) {
+            return undefined;
         }
-        const replicationEndpoints = this._config.replicationEndpoints
-            .map(endpoint => endpoint.site);
-        const locationConstraints =
-            Object.keys(this._config.locationConstraints);
-        if (locationConstraints.includes(storageClass)) {
-            if (this._prevStorageClass !== null &&
-                this._prevStorageClass !== storageClass) {
-                return errors.InvalidRequest.customizeDescription(
-                    'The storage class must be same for all rules when ' +
-                    'replicating objects to an external location');
+        const { replicationEndpoints } = this._config;
+        const sites = replicationEndpoints.map(endpoint => endpoint.site);
+        const storageClasses = destination.StorageClass[0].split(',');
+        const isValidStorageClass = storageClasses.every(storageClass => {
+            // Needed to determine how to parse destination bucket.
+            if (sites.includes(storageClass) ||
+                validStorageClasses.includes(storageClass)) {
+                this._hasScalityDestination = true;
+                return true;
             }
-            this._isExternalLocation = true;
-        }
-        if (!replicationEndpoints.includes(storageClass) &&
-            !locationConstraints.includes(storageClass) &&
-            !validStorageClasses.includes(storageClass)) {
+            if (validStorageClasses.includes(storageClass)) {
```

Comment:
i think the above `if` statement will always catch this condition

Reply:
This condition is for AWS compatible storage classes (e.g., 'STANDARD').

Reply:
Nice catch! Will update this.